src/external/hnswlib/hnswlib.h

```diff
@@ -0,0 +1,199 @@
+#pragma once
+#ifndef NO_MANUAL_VECTORIZATION
+#if (defined(__SSE__) || _M_IX86_FP > 0 || defined(_M_AMD64) || defined(_M_X64))
+#define USE_SSE
+#ifdef __AVX__
+#define USE_AVX
+#ifdef __AVX512F__
+#define USE_AVX512
+#endif
+#endif
+#endif
+#endif
+
+#if defined(USE_AVX) || defined(USE_SSE)
+#ifdef _MSC_VER
+#include <intrin.h>
+#include <stdexcept>
+void cpuid(int32_t out[4], int32_t eax, int32_t ecx) {
+    __cpuidex(out, eax, ecx);
+}
+static __int64 xgetbv(unsigned int x) {
+    return _xgetbv(x);
+}
+#else
+#include <x86intrin.h>
+#include <cpuid.h>
+#include <stdint.h>
+/*static void cpuid(int32_t cpuInfo[4], int32_t eax, int32_t ecx) {
```

Comment:
Wouldn't it be better to add it as a submodule and exclude these pieces from compilation by some realm build definition? This library seems to still actively receiving updates, would be better for future maintenance.